src/Parser.zig

```diff
@@ -4257,11 +4265,15 @@ fn mulExpr(p: *Parser) Error!Result {
                     else => unreachable,
                 };
             } else {
-                lhs.val = switch (lhs.val) {
-                    .unsigned => |v| .{ .unsigned = v % rhs.val.unsigned },
-                    .signed => |v| .{ .signed = @rem(v, rhs.val.signed) },
-                    else => unreachable,
-                };
+                if (lhs.val == .signed and lhs.val.signed == std.math.minInt(i64) and rhs.val.signed == -1) {
```

Comment:
I'm not sure if the Zig compiler couldn't figure out the type inference or if I was just doing something wrong, but I wasn't able to incorporate this check as part of an if expression in the `.signed` prong of the switch